docs/developer-docs/setup/best-practices/architecture.mdx

```diff
@@ -0,0 +1,96 @@
+# Architecture 
+
+## Overview 
+
+One of the most important decisions when designing a dapp is the determining the project's architecture. Three commonly used project architectures include:
+
+- **Single canister projects**: Contain a single canisters that contains the entire project's functionality. This may be a single service, a frontend webpage, or both within a single canister. Benefits of single canister projects include:
+
+    - Simplified management and maintenance.
+
+    - A lower cost than some multi-canister projects (though this depends on several factors). 
+
+    - Can be a good starting place for new developers as they learn about developing on ICP. 
+
+    Example use cases:
+
+    - Hosting a webpage or website.
+
+    - Hosting a simple service, such as a calculator.
+
+    - Hosting a simple game.
+
+    - Hosting a simple dapp with frontend and backend code together in a single canister. 
+
+- **Multi-canister projects**: Contains multiple canisters that divide the project's functions into several canisters with specific purposes. Multi-canister projects are the default project architecture generated by the `dfx new` command. The most common multi-canister architecture consists of a canister for the project's frontend assets and a canister for the dapp's backend code. Multi-canister projects are also commonly used when it makes sense to place different reusable services into separate canisters. Benefits of multi-canister projects include:
+
+    - Better scalability than single canister projects.
+    
+    - Upgrading individual services may be easier. 
+
+    - Canisters providing a reusable service can be configured as ['pullable'](/docs/current/developer-docs/setup/pulling-canister-dependencies) and re-used in other projects through [`dfx deps`](/docs/current/references/cli-reference/dfx-deps). 
```

Comment:
Should we also mention inter-canister calls and how they work?

Reply:
That's a great idea, I will make that revision 👍🏻